includes/class-shortcodes.php

```diff
@@ -0,0 +1,486 @@
+<?php
+namespace Activitypub;
+
+class Shortcodes {
+	/**
+	 * The post object we're currently working on
+	 *
+	 * @var WP_Post  $post    A WordPress Post Object
+	 */
+	private $post;
+
+	/**
+	 * Class constructor, registering WordPress then shortcodes
+	 *
+	 * @param WP_Post  $post    A WordPress Post Object
+	 */
+	public function __construct( $post = null ) {
+		if( $post == null ) {
+			$post = \get_post();
+		}
+
+		if( \is_object( $post ) && ! $post instanceof WP_POST ) {
+			$this->post = $post;
+		} else {
+			$this->post = false;
+		}
+
+		foreach( get_class_methods( $this ) as $shortcode ) {
```

Comment:
I like the idea, but I would still stick to a static class with an init function, because (if we remove the `post` var, see other comment) there is no need for an object context.

Reply:
I modified the code to just ignore the `init` function instead of looking for the double underscore, so the fundamental technique still works.